packages/forms/src/UIForm/fields/Code/Code.component.js

```diff
@@ -32,42 +45,79 @@ try {
 			this.props.onFinish(event, { schema: this.props.schema });
 		}
 
+		onKeyDown(event) {
+			if (event.keyCode === keyCode.codes.esc) {
+				const now = Date.now();
+				if (this.lastEsc && this.lastEsc - now < 1000) {
+					this.lastEsc = null;
+					this.ref.focus();
+					this.editor.textInput.getElement().setAttribute('tabindex', -1);
+				} else {
+					this.lastEsc = now;
+				}
+			} else {
+				this.lastEsc = null;
+			}
+		}
+
+		onBlur() {
+			this.editor.textInput.getElement().removeAttribute('tabindex');
+		}
+
 		onLoad(editor) {
-			editor.textInput.getElement().setAttribute('id', this.props.id);
+			this.editor = editor;
+			this.componentDidUpdate();
```

Comment:
im not sure this should be considered as a good idea to manually call for React's life-cycle function.

Reply:
It would be a little big to trigger a render to perform the action I want. Here I just want to add the attributes to the Ace textarea. And to avoid writing this twice, I called the componentDidUpdate that only have this code.
But I understand the issue, I'll extract that into another method, that will be called in cDU and here.